File: pkg/types/annotation.go
package types

type ENETag string

type AttributeName string

type Annotation struct {
	ENETag        ENETag        `json:"ENE"`
	AttributeName AttributeName `json:"attribute"`
	HTMLOffset    OffsetPair    `json:"html_offset"`
	TextOffset    OffsetPair    `json:"text_offset"`
	PageID        string        `json:"page_id"`
	Title         string        `json:"title"`
	LinkPageID    string        `json:"link_page_id"`
}

type OffsetPair struct {
	Start Offset `json:"start"`
	End   Offset `json:"end"`
	Text  string `json:"text"`
}

type Offset struct {
	LineID int `json:"line_id"`
	Offset int `json:"offset"`
}

File: python/src/types/annotation.py
from dataclasses import dataclass
from dataclasses_json import dataclass_json


@dataclass
class Offset:
    line_id: int
    offset: int


@dataclass
class OffsetPair:
    start: Offset
    end: Offset
    text: str


@dataclass_json
@dataclass
class Annotation:
    ENE: str
    attribute: str
    html_offset: OffsetPair
    text_offset: OffsetPair
    page_id: str
    title: str
    link_page_id: str = ""
